### PR TITLE
Plugins can send usage data

### PR DESCRIPTION
**Pull request by gturri**
They just need to register to the PLUGIN_USAGE_DATA event, and then to add
either a simple string, or an array of key / value. For example:

    function register(Doku_Event_Handler $controller) {
      $controller->register_hook('PLUGIN_USAGE_DATA', 'AFTER', $this, 'usage_data');
    }

    function usage_data(&$event){
      $event->data['my_plugin_name'] = 'my usage data';

      //or: $event->data['my_plugin_name'] = array ('k1' => 'v1', 'k2' => 'v2');
    }

### CODE REVIEW

File: lib/plugins/popularity/helper.php [hunk 1]
```diff
@@ -253,9 +253,26 @@ function _gather(){
         $data['php_exectime'] = $phptime;
         $data['php_extension'] = get_loaded_extensions();
 
+        // pluging usage data
+        $this->_add_plugin_usage_data($data);
+
         return $data;
     }
 
+    private function _add_plugin_usage_data(&$data){
```
Comment (splitbrain) [inline lib/plugins/popularity/helper.php]: should be protected

**Comment by splitbrain** [comment]
I think the event name is not adhering to our naming scheme https://www.dokuwiki.org/devel:events_list#naming_structure.  Though I am not 100% sure we have one for plugins. PLUGIN_POPULARITY_DATA_SETUP maybe?

**Comment by gturri** [comment]
Thanks for this feedback. Here is an update.

**Comment by gturri** [comment]
Thanks!

I just documented this feature on https://www.dokuwiki.org/devel:plugin_programming_tips#sending_popularity_data

**Comment by Klap-in** [comment]
Can you document the your event also at https://www.dokuwiki.org/devel:event:PLUGIN_POPULARITY_DATA_SETUP
so that it is added to https://www.dokuwiki.org/devel:events_list as well.
Thanks!

**Comment by gturri** [comment]
Done

**Pull request merged**